src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -224,6 +234,10 @@ public void setEntry(BibEntry entry) {
 
             setupToolBar();
         });
+
+        // listen to type changes for rebuilding the currently visible tab
+        typeListener = new EntryTypeListener(this.entry, bibDatabaseContext.getMode(), typeLabel, tabbed);
+        this.entry.getTypeProperty().addListener(typeListener);
```

Comment:
It is actually easier to use `Easybind.subcribe()` to listen to changes: you don't need a listener class and provide the code simply as a lambda expression. In this case, you get a subscription as the return value and call `unsubscribe` to, well, unsubscribe.

Reply:
Thanks! I've refactored the listener into an EasyBind subscription as you suggested. Slowly but steadily I learn more about programming JavaFX :)